### PR TITLE
Refactor gqcp_version to be a subfolder of gqcp during install.

### LINKED ISSUE #915: Documentation page on linking external projects to GQCP
[GQCD](https://github.com/GQCG-res/GQCD) was successfully linked to `GQCP`, so it would be useful to write a documentation page for future users on how to link external libraries using CMake.

### PR DESCRIPTION
**Pull request by dariatols**
**Short description**

Since `GQCP` could not find its own include folder, I think this had something to do with the include folder not being specified during installation.

EDIT: It **could** find its own include folder. I have adjusted the `gqcp_version` library to be a part of `gqcp` to make installation less dubious.

** Related issues**
Closes #914.
Closes #915.

**To do**

- [x] (if applicable, when creating new source files): don't forget to update the Doxygen file, the collective `gqcp.hpp` include header and the CMake files
- [x] Add other smaller task to be completed as a Markdown task list

### CODE REVIEW

**Comment by lelemmen** [comment]
@dariatols Could you add a test that verifies this behavior?

**Comment by dariatols** [comment]
@lelemmen This PR is not done yet, as I am not able to let the installed `GQCP` find its own header files yet. I'll convert the PR to 'Ready for review' once everything works and an additional test is included!

**Comment by dariatols** [comment]
After all, `GQCP` could find its own include folder. I have adjusted the gqcp_version library to be a part of gqcp to make installation less dubious.

**Comment by lelemmen** [comment]
@dariatols Do our current tests test for this behavior? Or is #914 required?

**Comment by dariatols** [comment]
@lelemmen Our current tests do not test for this behavior yet as #914 is required.

**Comment by lelemmen** [comment]
Ok. Once a relevant test case is included, this PR can be merged. Could you also write a little documentation page detailing how to link the `C++` library to an external CMake project?

**Comment by guacke** [comment]
> After all, `GQCP` could find its own include folder. I have adjusted the gqcp_version library to be a part of gqcp to make installation less dubious.

Can you frame this more? How did you solve this? And why is the current installation 'dubious'? I have added a linker test in PR #914, but this test is currently failing. The underlying error _seems_ to be related to the recent CMake version changes, that were implemented without unit tests.

**Comment by dariatols** [comment]
I had a bug in `GQCD` where `GQCP` could not find its own include headers, but this was due to a circular `include` on my behalf. The current installation of `GQCP` is now less dubious since the installation now places the version folder in `/usr/local/miniconda3/include/gqcp` instead of just `/usr/local/miniconda3/include`.

**Comment by dariatols** [comment]
@lelemmen @guacke #914 is fixed now, there was a redundant `version.hpp` include in `gqcp.hpp` that I forgot to remove.

**Comment by lelemmen** [comment]
@dariatols Please include the changes that are proposed in #914 into this PR, so that this PR has CI-validity.

Also, please resolve #915 in this PR as well.

**Comment by dariatols** [comment]
**To do**
Write documentation on how to link with `GQCP::gqcp`.

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -24,5 +24,12 @@ jobs:
...
-        cmake .. -DBUILD_TESTS=TRUE -DBUILD_BENCHMARKS=TRUE -DBUILD_PYTHON_BINDINGS=TRUE -DCMAKE_INSTALL_PREFIX=~/.local -DCMAKE_PREFIX_PATH=/usr/local/miniconda3 -DPYTHON_EXECUTABLE=/usr/local/miniconda3/bin/python3 -DPYTHON_LIBRARY=/usr/local/miniconda3/lib/libpython3.8m.a
+        cmake .. -DBUILD_TESTS=TRUE -DBUILD_BENCHMARKS=TRUE -DBUILD_PYTHON_BINDINGS=TRUE -DCMAKE_INSTALL_PREFIX=/usr/local/miniconda3 -DCMAKE_PREFIX_PATH=/usr/local/miniconda3 -DPYTHON_EXECUTABLE=/usr/local/miniconda3/bin/python3 -DPYTHON_LIBRARY=/usr/local/miniconda3/lib/libpython3.8m.a
         make -j2 VERBOSE=1 && env CTEST_OUTPUT_ON_FAILURE=1 make test && make install
+    - name: Link test
+      run: |
+        cd tools/link-test
+        mkdir build && cd build
+        cmake .. -DCMAKE_MODULE_PATH=/usr/local/miniconda3/cmake
+        make -j2 VERBOSE=1 && ./link
+
```
Comment (lelemmen) [inline .github/workflows/build-and-test.yml:35]:
```suggestion

```

File: tools/link-test/link.cpp [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+ */
+
+#include <gqcp/gqcp.hpp>
+#include <stdio.h>
+
+int main() {
+    GQCP::Nucleus nucleus (GQCP::elements::elementToAtomicNumber("N"), -1/2, 0, 0);
+    std::cout << "Link successful." << std::endl;
+    return 0;
+}
```
Comment (lelemmen) [inline tools/link-test/link.cpp:12]:
```suggestion
}

```

**Review by lelemmen — APPROVED** [review]

**Pull request merged**